mempool/mempool.go

```diff
@@ -902,60 +905,58 @@ func (mp *TxPool) maybeAcceptTransaction(tx *dcrutil.Tx, isNew, rateLimit, allow
 		}
 	}
 
-	// Handle stake transaction double spending exceptions.
-	if (txType == stake.TxTypeSSGen) || (txType == stake.TxTypeSSRtx) {
-		if txType == stake.TxTypeSSGen {
-			ssGenAlreadyFound := 0
-			for _, mpTx := range mp.pool {
-				if mpTx.Type == stake.TxTypeSSGen {
-					if mpTx.Tx.MsgTx().TxIn[1].PreviousOutPoint ==
-						msgTx.TxIn[1].PreviousOutPoint {
-						ssGenAlreadyFound++
-					}
+	// Aside from a few exceptions for votes and revocations, the transaction
+	// may not use any of the same outputs as other transactions already in
+	// the pool as that would ultimately result in a double spend.  This
+	// check is intended to be quick and therefore only detects double
+	// spends within the transaction pool itself.  The transaction could
+	// still be double spending coins from the main chain at this point.
+	// There is a more in-depth check that happens later after fetching the
+	// referenced transaction inputs from the main chain which examines the
+	// actual spend data and prevents double spends.
+	if !isVote && !isRevocation {
+		err = mp.checkPoolDoubleSpend(tx, txType)
+		if err != nil {
+			return nil, err
+		}
+	} else if isVote {
+		ssGenAlreadyFound := 0
+		for _, mpTx := range mp.pool {
+			if mpTx.Type == stake.TxTypeSSGen {
+				if mpTx.Tx.MsgTx().TxIn[1].PreviousOutPoint ==
+					msgTx.TxIn[1].PreviousOutPoint {
+					ssGenAlreadyFound++
 				}
-				if ssGenAlreadyFound > maxSSGensDoubleSpends {
+			}
+			if ssGenAlreadyFound > maxSSGensDoubleSpends {
+				str := fmt.Sprintf("transaction %v in the pool "+
```

Comment:
The error messages use`sstx`, `ssgen` and `ssrtx` which we are moving away from. Consider using `ticket`, `vote` and `revocation` instead.

Reply:
Good point.  I didn't mess with cleaning up the errors as I was focused on the variables for upcoming PRs, but it makes sense to do so as a part of this PR.